tests/unit/lib/functions/registry.test.js

```diff
@@ -0,0 +1,33 @@
+const { mkdirSync, mkdtempSync, writeFileSync } = require('fs')
+const { tmpdir } = require('os')
+const { join } = require('path')
+
+const test = require('ava')
+const sinon = require('sinon')
+
+const { FunctionsRegistry } = require('../../../../src/lib/functions/registry')
+
+test('registry should only pass functions config to zip it ship it', async (t) => {
+  const projectRoot = mkdtempSync(join(tmpdir(), 'functions-project-root'))
```

Comment:
We should clean up this directory at the end of the test.

Reply:
Done.  Can you help me understand why this only works when I await the `rmdir()` call?  Does the javascript process end before the actual async function is run otherwise?

Reply:
I don't follow. Can you share an example of what works and what doesn't?

Reply:
Yes if you do not await `rmdir` Node.JS will not wait for it and just shut down.